docs/5.rest-api-endpoints.md

```diff
@@ -0,0 +1,61 @@
+
+**Endpoints**
+
+1.  **List Features:** `GET /wp/v2/features`
+    *   Retrieves a paginated list of all registered and *eligible* features.
+    *   Supports query parameters for filtering (see `get_collection_params()` in the controller):
+        *   `page`: Page number (default: 1).
+        *   `per_page`: Items per page (default: 10, max: 100).
+        *   `fields`: Comma-separated list of fields to include (e.g., `id,name,description`).
+        *   `type[]`: Filter by type (e.g., `type[]=resource&type[]=tool`).
+        *   `categories[]`: Filter by category slug (e.g., `categories[]=core&categories[]=post`).
+        *   `location[]`: Filter by location (e.g., `location[]=server`).
+        *   `search`: Search term.
+    *   Response includes pagination headers (`X-WP-Total`, `X-WP-TotalPages`) and `Link` headers for `prev`/`next`.
+    *   **Example:** `GET /wp-json/wp/v2/features?per_page=5&type[]=tool&categories[]=woocommerce`
+
+2.  **List Categories:** `GET /wp/v2/features/categories`
+    *   Retrieves a list of all registered feature categories.
+    *   **Example:** `GET /wp-json/wp/v2/features/categories`
+
+3.  **Get Specific Category:** `GET /wp/v2/features/categories/{id}`
+    *   Retrieves details for a single category by its slug.
+    *   `{id}`: The category slug (e.g., `core`, `woocommerce`).
+    *   **Example:** `GET /wp-json/wp/v2/features/categories/woocommerce`
+
+4.  **Get Specific Feature:** `GET /wp/v2/features/{feature-id}`
+    *   Retrieves details for a single feature by its full ID (including type prefix).
+    *   `{feature-id}`: The full feature ID (e.g., `resource-users/me`).
+    *   **Example:** `GET /wp-json/wp/v2/features/resource-users/me`
+
+5.  **Run Feature:** `POST|GET /wp/v2/features/{feature-id}/run`
+    *   Executes the specified feature.
+    *   `{feature-id}`: The full feature ID (e.g., `tool-myplugin/update-option`, `resource-myplugin/active-theme`).
+    *   **Method:** Use `POST` for `tool` features, `GET` for `resource` features (though `POST` often works for resources too if sending context).
+    *   **Request Body (JSON for POST):** Contains the context/arguments for the feature.
+        ```json
+        {
+          "context": {
+            "option_name": "blogdescription",
+            "option_value": "Updated via REST"
+          }
+        }
+        ```
+    *   **Query Parameters (for GET resources with params like REST aliases):** Add parameters to the URL if the feature expects them (often derived from `rest_alias` routes). Example: `GET /wp-json/wp/v2/features/resource-post/run?context[id]=1` (although the controller doesn't explicitly parse `context` from query params currently, a direct `call` or custom feature might). REST Aliases map URL params automatically.
```

Comment:
I'm changing this a bit in my branch for a couple different reasons:

I've had the agent use `context` as an input for some tools. We don't actually support getting this from query strings at the moment, and not all `/run` endpoints accept `POST`.

Instead, you can pass inputs as query parameters for GET requests, or as the body for POST requests.

I can update the docs in my branch after merging this and then rebasing.